lib/ts/recipe/emailpassword/utils.ts

```diff
@@ -114,25 +119,58 @@ function validateAndNormaliseResetPasswordUsingTokenConfig(
     };
 }
 
-function normaliseSignInFormFields(formFields: NormalisedFormField[]) {
-    return formFields
-        .filter((filter) => filter.id === FORM_FIELD_EMAIL_ID || filter.id === FORM_FIELD_PASSWORD_ID)
-        .map((field) => {
-            return {
-                id: field.id,
-                // see issue: https://github.com/supertokens/supertokens-node/issues/36
-                validate: field.id === FORM_FIELD_EMAIL_ID ? field.validate : defaultValidator,
-                optional: false,
-            };
+function normaliseSignInFormFields(formFields?: TypeInputFormField[]) {
+    let normalisedFormFields: NormalisedFormField[] = [];
+    if (formFields !== undefined) {
+        formFields.forEach((field) => {
+            if (field.id === FORM_FIELD_PASSWORD_ID) {
+                normalisedFormFields.push({
+                    id: field.id,
+                    validate: field.validate === undefined ? defaultPasswordValidator : field.validate,
+                    optional: false,
+                });
+            } else if (field.id === FORM_FIELD_EMAIL_ID) {
+                normalisedFormFields.push({
+                    id: field.id,
+                    validate: field.validate === undefined ? defaultEmailValidator : field.validate,
+                    optional: false,
+                });
+            } else {
+                normalisedFormFields.push({
+                    id: field.id,
+                    validate: field.validate === undefined ? defaultValidator : field.validate,
+                    optional: field.optional === undefined ? false : field.optional,
+                });
+            }
         });
+    }
+    if (normalisedFormFields.filter((field) => field.id === FORM_FIELD_PASSWORD_ID).length === 0) {
+        // no password field give by user
+        normalisedFormFields.push({
+            id: FORM_FIELD_PASSWORD_ID,
+            validate: defaultPasswordValidator,
+            optional: false,
+        });
+    }
+    if (normalisedFormFields.filter((field) => field.id === FORM_FIELD_EMAIL_ID).length === 0) {
+        // no email field give by user
+        normalisedFormFields.push({
+            id: FORM_FIELD_EMAIL_ID,
+            validate: defaultEmailValidator,
```

Comment:
This should fall back to the (normalized) email field validator for sign-up to avoid breaking apps.